src/main/kotlin/ar/edu/unahur/obj2/semillasAlViento/Parcela.kt

```diff
@@ -1,45 +1,40 @@
 package ar.edu.unahur.obj2.semillasAlViento
 
 class Parcela(val ancho: Int, val largo: Int, val horasSolPorDia: Int) {
-  val plantas = mutableListOf<Planta>()
-  var cantidadPlantas = 0
+    val plantas = mutableListOf<Planta>()
+    fun superficie() = ancho * largo
+    fun cantidadMaximaPlantas() =
+            if (ancho > largo) this.superficie() / 5 else this.superficie() / 3 + largo
 
-  fun superficie() = ancho * largo
-  fun cantidadMaximaPlantas() =
-    if (ancho > largo) ancho * largo / 5 else ancho * largo / 3 + largo
-
-  fun plantar(planta: Planta) {
-    if (cantidadPlantas == this.cantidadMaximaPlantas()) {
-      println("Ya no hay lugar en esta parcela")
-    } else if (horasSolPorDia > planta.horasDeSolQueTolera() + 2) {
-      println("No se puede plantar esto acá, se va a quemar")
-    } else {
-      plantas.add(planta)
-      cantidadPlantas += 1
+    fun plantar(planta: Planta) {
+        when {
+            this.plantas.size == this.cantidadMaximaPlantas() -> {
+                throw Exception("Ya no hay lugar en esta parcela")
+            }
+            this.horasSolPorDia >= planta.horasDeSolQueTolera() + 2 -> {
+                throw Exception("No se puede plantar esto acá, se va a quemar")
+            }
+            else -> {
+                plantas.add(planta)
+            }
+        }
     }
-  }
-}
 
-class Agricultora(val parcelas: MutableList<Parcela>) {
-  var ahorrosEnPesos = 20000
+    fun esSemillera() = if (this.plantas.isNotEmpty()) this.plantas.all { it.daSemillas() } else false
```

Comment:
La definición del `all` alcanzaba, es correcto en términos lógicos decir que una parcela sin plantas es semillera. De todas formas, si en otro momento quisieras hacer algo así, podés usar el operador `and`:

```suggestion
    fun esSemillera() = this.plantas.isNotEmpty() && this.plantas.all { it.daSemillas() }
```